app/src/main/java/org/mozilla/fenix/tabhistory/TabHistoryAdapter.kt

```diff
@@ -18,23 +19,23 @@ data class TabHistoryItem(
 
 class TabHistoryAdapter(
     private val interactor: TabHistoryViewInteractor
-) : RecyclerView.Adapter<TabHistoryViewHolder>() {
-
-    var historyList: List<TabHistoryItem> = emptyList()
-        set(value) {
-            field = value
-            notifyDataSetChanged()
-        }
+) : ListAdapter<TabHistoryItem, TabHistoryViewHolder>(DiffCallback) {
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TabHistoryViewHolder {
-        val view =
-            LayoutInflater.from(parent.context).inflate(R.layout.history_list_item, parent, false)
+        val view = LayoutInflater.from(parent.context)
+            .inflate(R.layout.history_list_item, parent, false)
```

Comment:
I shouldn't have used this layout because it contains the header and button views we use in the history fragment. It also wraps the whole `LibrarySiteItemView` in another LinearLayout. I think it would be best to create a new layout file that just contains the `LibrarySiteItemView` with the same attributes that `history_layout` has. And this becomes something like

```suggestion
            .inflate(R.layout.tab_history_list_item, parent, false)
```